redaxo/src/core/lib/sql/sql.php

```diff
@@ -415,14 +415,23 @@ public function setQuery($query, array $params = [], array $options = [])
         }
 
         try {
-            $this->stmt = rex_timer::measure(__METHOD__, static function () use ($pdo, $query) {
-                return $pdo->query($query);
+            $this->stmt = rex_timer::measure(__METHOD__, function () use ($pdo, $query) {
+                error_clear_last();
+
+                if (false !== $stmt = @$pdo->query($query)) {
+                    return $stmt;
+                }
+
+                if ($error = error_get_last()) {
+                    throw new rex_sql_exception('Error while executing statement "' . $query . '": ' . $error['message'], null, $this);
+                }
+                throw new rex_sql_exception('Error while executing statement "' . $query . '".', null, $this);
```

Comment:
sollten wir hier die errors von pdo nehmen, statt via error_get_last()?

https://www.php.net/manual/en/pdo.errorinfo.php
https://www.php.net/manual/en/pdo.errorcode.php

Reply:
Das Problem ist: Normale SQL-Fehler bekommen wir ja als Exception und fangen wir sowieso schon ab.
Und ich hab jetzt nur die Info von heute aus Slack, wo es eine Warning gab und dann den false-Wert. Ich weiß nicht, ob die Warning in dem Fall in errorinfo drin gewesen wäre.
Und ich weiß eben auch nicht, wann genau die Kombi Warning+false auftritt trotz Exception-Mode in PDO.

Reply:
ok, hab oben nen kommentar ergänzt und dann könnnen wir.